ansible/examples/builder.yml

```diff
@@ -44,6 +43,62 @@
       tags:
         always
 
+    - name: Stop unattended-upgrades to avoid conflicts
+      systemd:
+        name: unattended-upgrades
+        state: stopped
+      when: ansible_os_family == "Debian"
+      ignore_errors: yes
+
+    - name: Wait for APT lock to be released
+      command: "timeout 300 bash -c 'while fuser /var/lib/apt/lists/lock /var/cache/apt/archives/lock /var/lib/dpkg/lock-frontend >/dev/null 2>&1; do sleep 5; done'"
```

Comment:
I see that this is recommended in a few places; this blog entry makes the point that it's got some problems, and maybe offers a better solution with apt config variables:  https://blog.sinjakli.co.uk/2021/10/25/waiting-for-apt-locks-without-the-hacky-bash-scripts/.  I see that the apt module supports dpkg_options: https://docs.ansible.com/ansible/latest/collections/ansible/builtin/apt_module.html#parameters   WDYT?

Reply:
These options don’t address the lock issue, but for "Upgrade all packages (Debian)" task I think of adding force-confold and force-confdef, this will ensure old configuration files are kept (force-confold) and defaults are used for new configs (force-confdef), it will reduce interactive prompts

Reply:
For the lock task, I will use wait_for instead of using the fuser

Reply:
You're right, I thought it's just waiting without checking if the lock files exist or not, I have added the lock_timeout to the packages upgrade tasks and I also added those options force-confold,force-confdef as mentioned in my previous comment

Reply:
ok, well, the lock_timeout parameter is different yet, and is not implemented with DPkg::Lock::Timeout, but maybe it is good enough.  The module implements its own timeout loop.